### PR TITLE
ISPN-10511 Docs: Cache modes, clustered caches, partition handling

### PR DESCRIPTION
**Pull request by oraNod**
https://issues.jboss.org/browse/ISPN-10511 

Moving the content of the current clustering title to a new title called "Infinispan Cache Guide". The current clustering title doesn't fit the content.

### CODE REVIEW

File: documentation/src/main/asciidoc/titles/partitioning/partitioning.html [hunk 1]
```diff
@@ -0,0 +1,997 @@
+<!DOCTYPE html>
```
Comment (danberindei) [inline documentation/src/main/asciidoc/titles/partitioning/partitioning.html]: Included by mistake?

File: documentation/src/main/asciidoc/topics/con_cache_containers.adoc [hunk 2]
```diff
@@ -1,16 +1,17 @@
...
-
-[source,xml,options="nowrap",subs=attributes+]
-----
-include::config_examples/ispn_jgroups_transport.xml[]
-----
+//dnaro: need to clarify this note.
+[NOTE]
+====
+{brandname} servers let you specify named caches as default dynamically, rather
+than providing implicit default caches.
```
Comment (danberindei) [inline documentation/src/main/asciidoc/topics/con_cache_containers.adoc]: The intent here was to contrast the server configuration, where the default cache is selected with the `<cache-container default-cache="<cache>">`, with the old embedded configuration, where we had a `<default>` element:

https://github.com/infinispan/infinispan/blob/6.0.x/core/src/test/resources/configs/batching.xml#L6

Embedded mode switched to the server style config in 7.0, so there's no point in mentioning the implicit default cache any more.

**Comment by tristantarrant** [comment]
Needs rebase

**Comment by tristantarrant** [comment]
Merged

**Pull request merged**